### PR TITLE
Simplify type check

### PR DESCRIPTION
**Pull request by RGates94**
Simplifies control flow and removes unneeded code in typecheck.rs.

### CODE REVIEW

File: src/compile/typecheck.rs [hunk 1]
```diff
@@ -17,7 +17,7 @@ use crate::stringtable::{StringId, StringTable};
 use crate::uint256::Uint256;
 use std::collections::{HashMap, HashSet};
 
-type TypeTable<'a> = HashMap<usize, &'a Type>;
+type TypeTable<'a> = HashMap<usize, Type>;
```
Comment (edfelten) [inline src/compile/typecheck.rs:20]: Do we really need the <`a> on TypeTable?

**Review by edfelten — CHANGES_REQUESTED** [review]
One tiny issue.

File: src/compile/typecheck.rs (hunk 1 above)
Comment (RGates94) [reply src/compile/typecheck.rs:20]: Good catch, its no longer needed.

**Review by edfelten — APPROVED** [review]
Looks good.

**Pull request merged**